### DISCUSSION TITLE
adapting forme to bootstrap 5

### DISCUSSION THREAD

**Discussion by okliv** [jeremyevans/forme#101] [General]
Hi!

What will you advice to override/rewrite to make forme suit well with bootstrap 5 (or any)?

Thanks

**Comment by jeremyevans** [comment]
I would start with the existing bootstrap 3 support and make the necessary adjustments.

**Reply by okliv** [reply]
will it be enough to `cp lib/forme/bs3.rb lib/forme/bs5.rb` and modify it?

**Reply by jeremyevans** [reply]
Yep, that sounds like the best way.

**Comment by janko** [comment]
@okliv How far did you get with this? I wanted this recently for smoother screencasting, and put something together in couple of hours that seems to be working, and was thinking of submitting a pull request.

**Reply by okliv** [reply]
first quick look&try didn't bring me final result, so, it is in my schedule. for now i re-implemented my forms on views layer (instead of initial model layer) and make it look nice with raw html and *_wrapper modifiers

so, if you have working solution it would be great to try it

**Reply by janko** [reply]
Here it is, I included some examples on how it can be used – https://gist.github.com/janko/2bf68dfe36b9886d5a0ed9b3d37df49f. Let me know how it works for you 🙂 

It's significantly smaller than the Bootstrap 3 implementation, partly due to markup changes in Bootstrap 5 that simplify the logic, but also because I was able to reuse more code.

**Reply by okliv** [reply]
nice! actually, main reason for me were [switches](https://getbootstrap.com/docs/5.0/forms/checks-radios/#switches) instead of checkboxes. will try this tomorrow, thanks!

**Reply by janko** [reply]
Yeah, those should work, you just pass `wrapper_attr: { class: "form-switch" }` to the input.
